src/lib.rs

```diff
@@ -260,6 +261,7 @@ impl<'a> Authentication<'a> {
 
 mod error;
 pub mod tcp;
+pub mod tcp_socks4;
```

Comment:
I prefer to put the socks4 types directly in the `tcp` mod.

To better organize the code, maybe we can move the current socks5 code to tcp/socks5 and put the sock4 code in tcp/socks4. Then, re-export the useful types to the tcp mod for convenience.

Reply:
Yep seems fine, can I do it as part of other PR? or this one itself.

Reply:
It's okay to just to do it in this PR.

Reply:
Done.